addons/purchase_requisition/wizard/purchase_requisition_create_alternative.py

```diff
@@ -80,15 +80,27 @@ def _get_alternative_values(self):
             'origin': self.origin_po_id.origin,
         }
         if self.copy_products and self.origin_po_id:
-            vals['order_line'] = [Command.create(self._get_alternative_line_value(line)) for line in self.origin_po_id.order_line]
+            product_tmpl_ids_with_description = {
+                product_template[0].id
+                for product_template in self.env['product.supplierinfo']._read_group(
+                    domain=[
+                        ('product_tmpl_id', 'in', self.origin_po_id.order_line.product_id.product_tmpl_id.ids),
+                        ('partner_id', '=', self.partner_id.id),
+                        '|', ('product_code', '!=', False), ('product_name', '!=', False)
+                    ],
+                    groupby=['product_tmpl_id'],
+                )
+            }
```

Comment:
```suggestion
            product_tmpl_ids_with_description = set(self.env['product.supplierinfo'].search(
                domain=[
                    ('product_tmpl_id', 'in', self.origin_po_id.order_line.product_id.product_tmpl_id.ids),
                    ('partner_id', '=', self.partner_id.id),
                    '|', ('product_code', '!=', False), ('product_name', '!=', False)
                ],
            ).product_tmpl_id.ids)
```
Seems a bit easier to read. What do you think ?

Reply:
Oh yes, you're right — it is definitely cleaner and easier to read.

But performance-wise, I think `_read_group` is more efficient than `search` as it returns only the grouped data that matches the domain, rather than loading full records into memory. whereas search fetches all matched records into memory.

and I think @ticodoo also suggest it before for the same purpose here : https://github.com/odoo/odoo/pull/189699#discussion_r1959520819

Looking forward to your thoughts as well, @Whenrow 

Thanks !! 😊

Reply:
@chha-odoo It requires 2 queries and fill the cache with all fields of `product.supplier.info`. You can also use `search_fetch` if you want to increase performance

Reply:
> @chha-odoo It requires 2 queries and fill the cache with all fields of `product.supplier.info`. You can also use `search_fetch` if you want to increase performance

Hello @Whenrow ,
I've updated the code and use `search_fetch` as it helps to improve performance.
Thanks !! 😊